src/cli/self_update.rs

```diff
@@ -344,6 +344,8 @@ pub fn install(
 
         if !opts.no_modify_path {
             do_add_to_path()?;
+            #[cfg(windows)]
+            do_add_to_programs()?;
```

Comment:
There are desktop environments on Unix OSes too that have similar app registration mechanisms that we should perhaps be using. Rather than cfg windows, perhaps just a register_program() call, and a stub for unix platforms, and something real for windows?